test/node/tools/utils.js

```diff
@@ -176,3 +176,17 @@ const getSymbolFrom = function (target, symbolName, assertExists) {
 };
 
 exports.getSymbolFrom = getSymbolFrom;
+
+module.exports.byStrings = (a, b) => {
+  const res = `${a}`.localeCompare(`${b}`);
+  return res < 0 ? -1 : res > 0 ? 1 : 0;
```

Comment:
Nothing wrong with this, but why not …?

```suggestion
  return `${a}`.localeCompare(`${b}`);
```

Reply:
It's copy pasta-ed from our driver where this is written in TS and localeCompare returns number not the required union of `0 | 1 | -1` so that's why we have this little coercion ternary. it would clean it up but we've copied this function in 3 places (its in mongodb-legacy too) so might as well keep it identical despite the extra logic

Reply:
Interesting! I’d be curious to know more, the TS definition for `Array.prototype.sort()` is `sort(compareFn?: (a: T, b: T) => number): this`, so it should work out fine to use `localeCompare` directly in the driver too? But obviously this is getting into more detail than necessary :)

Reply:
No totally good to keep me honest here about motivations. I think this strict union requirement was inspired by the bug we had to fix across all drivers relating to setVersion/electionId comparison, for [our implementation in particular](https://github.com/mongodb/node-mongodb-native/blob/main/src/sdam/topology_description.ts#L377-L381) we have this carefully written logic for determining greaterThan/lessThan/equal that's easier to follow when you can name variable representing `=== 0` or `=== -1`. (of course we could've still made those variables using `> 0` or `< -1`) Now I've just brought that TS requirement with me when writing compare functions for other use cases since it proved useful there. It is a burden brought on by own design 😭 😅 

Personal anecdote, compare functions and I have never been friends, going back to my OS class I wrote my [compare function to only return true or false](https://gitlab.com/moon-project/moon/-/blob/develop/libc/string.c#L18-27) 🫣 since (for whatever reason) I believed that to be more logical than `-1,1,0` after all isn't equality all we care about?? (of course not) thus breaking our libc implementation 😭